Update nix index

lehins opened the pull request:
Updates all sources: `hackage.nix`, `haskell.nix`, `iohk-nix` and `ormolu`

lehins committed:
Update nix index and account for new version of libraries in byron

lehins committed:
Update the rest of nix deps as well as ormolu

lehins committed:
Ormolize with ormolu-0.4.0.0

nc6 commented on eras/byron/ledger/impl/test/Test/Cardano/Chain/Delegation/Model.hs:
```diff
@@ -74,8 +74,11 @@ newtype SignalSDELEG (v :: Type -> Type)
 
-instance HTraversable SignalSDELEG where
-  htraverse _ s = pure (coerce s)
+instance FunctorB SignalSDELEG where
```
From `barbies`? I didn't even know we used that...

nc6 reviewed (approved):
LGTM

lehins committed:
Fix golden test to compare on `Value` instead of raw JSON

nc6 committed:
Add nixpkgs pin.

This is needed to include a later libsecp256k1 version with support for
Schnorr signatures.

This change was made with the following:
```
niv add nixos/nixpkgs -v 1882c6b7368fd284ad01b0a5b5601ef136321292
```

The revision was picked to match that in base: https://github.com/input-output-hk/cardano-base/commit/1b8ee28677f5f3bb55a852a982357d0e48e14476

nc6 committed:
Use the correct libsodium version.

Note that we now also require secp256k1, so we need to keep that in the
pkgconfig.

lehins committed:
Remove Windows and MacOS stabs on GA and fix Ubuntu

nc6 committed:
Update CDDL tool

nc6 committed:
Update cbor-diag tool.
